Sources/URI/Model/URI.swift

```diff
@@ -20,35 +20,29 @@ public struct URI {
     public var scheme: String
 
     // https://tools.ietf.org/html/rfc3986#section-3.2.1
-    public let userInfo: UserInfo?
+    public var userInfo: UserInfo?
```

Comment:
what's the reason behind making all of these publicly settable? I think internally variable or constants is probably better